pkg/server/completion.go

```diff
@@ -185,8 +196,67 @@ func createCompletionItem(label, detail string, kind protocol.CompletionItemKind
 		},
 		InsertText: insertText,
 	}
+
+	// Remove leading `.` character when quoting label
+	if !mustNotQuoteLabel {
+		item.TextEdit = &protocol.TextEdit{
+			Range: protocol.Range{
+				Start: protocol.Position{
+					Line:      position.Line,
+					Character: position.Character - 1,
+				},
+				End: protocol.Position{
+					Line:      position.Line,
+					Character: position.Character,
+				},
+			},
+			NewText: insertText,
+		}
+	}
+
+	return item
+}
+
+// Start - Copied from go-jsonnet/internal/parser/lexer.go
+
+func isUpper(r rune) bool {
+	return r >= 'A' && r <= 'Z'
+}
+func isLower(r rune) bool {
+	return r >= 'a' && r <= 'z'
+}
+func isNumber(r rune) bool {
+	return r >= '0' && r <= '9'
+}
+func isIdentifierFirst(r rune) bool {
+	return isUpper(r) || isLower(r) || r == '_'
+}
+func isIdentifier(r rune) bool {
+	return isIdentifierFirst(r) || isNumber(r)
+}
+func IsValidIdentifier(str string) bool {
+	if len(str) == 0 {
+		return false
+	}
+	for i, r := range str {
+		if i == 0 {
+			if !isIdentifierFirst(r) {
+				return false
+			}
+		} else {
+			if !isIdentifier(r) {
+				return false
+			}
+		}
+	}
+	// Ignore tokens for now, we should ask upstream to make the formatter a public package
+	// so we can use go-jsonnet/internal/formatter/pretty_field_names.go directly.
+	// return getTokenKindFromID(str) == tokenIdentifier
+	return true
 }
 
+// End - Copied from go-jsonnet/internal/parser/lexer.go
```

Comment:
I've made a copy from upstream code as the formatter that has this logic is in an `internal/` package. We should probably ask to make the formatter a public package.

Reply:
Ignoring the tokens because that would be too much to copy and maintain.